readthedocs/doc_builder/backends/sphinx.py

```diff
@@ -395,6 +397,57 @@ def build(self):
             raise BuildEnvironmentError('No TeX files were found')
 
         # Run LaTeX -> PDF conversions
+        if self.project.has_feature(Feature.USE_PDF_LATEXMK):
+            return self._build_latexmk(cwd, latex_cwd)
+
+        return self._build_pdflatex(tex_files, latex_cwd)
+
+    def _build_latexmk(self, cwd, latex_cwd):
+        # These steps are copied from the Makefile generated by Sphinx >= 1.6
+        # https://github.com/sphinx-doc/sphinx/blob/master/sphinx/texinputs/Makefile_t
+        latex_path = Path(latex_cwd)
+        images = []
+        for extension in ('png', 'gif', 'jpg', 'jpeg'):
+            images.extend(latex_path.glob(f'*.{extension}'))
+
+        # FIXME: instead of checking by language here, what we want to check if
+        # ``latex_engine`` is ``platex``
+        pdfs = []
+        if self.project.language == 'ja':
+            pdfs = latex_path.glob('*.pdf')
```

Comment:
Why is this only required by ja lang? Don't we already run this whole function under a feature flag?

Reply:
Japanese language is the only one that requires this extra step. I don't know _exactly_ why but most of the documentation that I read differentiate this language from the others. I suppose it's because it mix kanji (Chinese) with its own symbols.

I took this step from the Makefile generated by Sphinx when the language is Japanese.

Reply:
The Japanese language uses `latex+dvipdfmx` path. Until TeXLive 2015 release, `extractbb` step was needed. But it is not needed with newer `dvipdfmx` binary. Here is a quote of an [exchange](https://tug.org/pipermail/tex-live/2017-January/039545.html) I had in 2017 on TeXLive mailing list:

> > > Can someone confirm that since TL2014 there is no need
> > > on Unix-like systems to prepare beforehand .xbb
> > > files for graphics inclusion of .png, .jpeg, ..., 
> > > when doing latex+dvipdfmx or platex+dvipdfmx ?
> > > (and that it was either needed before, or at least
> > > shell-escape had to be enabled) ?
> > > 
> > 
> 
> > I remember xbb files were necessary for TL2014 or earlier.
> > Japanese people has been heavily using dvipdfmx, so they added
> > extractbb to their local texmf.cnf by hand
> > 
> > ...
> 

> 
> Thanks for the information ! This is for support
> of some other software, now I know they can avoid
> the explicit extractbb calls but under assumption
> of a TL2015 or more recent install,

Thus Sphinx may soon drop the `extractbb` and make the Japanese set-up in Makefile less singular: indexe 2.0 release requires "LaTeX builder now depends on TeX Live 2015 or above." but we forgot to actually remove the `extractbb` dependency as we don't want either to break too easily user projects with older TeX installations.

Reply:
I have prepared https://github.com/sphinx-doc/sphinx/pull/6189 for Sphinx to remove the extra handling of image files for Japanese language by `extractbb` as it is superfluous for TL2015 or later based TeX distribution. Not sure it will be merged during 2.x series.